tests/phpunit/tests/rest-api/rest-server.php

```diff
@@ -913,14 +913,39 @@ public function test_link_embedding_cache_collection() {
 	 * Ensure embedding is a no-op without links in the data.
 	 */
 	public function test_link_embedding_without_links() {
-		$data   = array(
-			'untouched' => 'data',
+		$data   = new WP_REST_Response(
+			array(
+				'untouched' => 'data',
+			)
 		);
-		$result = rest_get_server()->embed_links( $data );
+		$result = rest_get_server()->response_to_data( $data, true );
 
-		$this->assertArrayNotHasKey( '_links', $data );
-		$this->assertArrayNotHasKey( '_embedded', $data );
-		$this->assertSame( 'data', $data['untouched'] );
+		$this->assertArrayNotHasKey( '_links', $result );
+		$this->assertArrayNotHasKey( '_embedded', $result );
+		$this->assertSame( 'data', $result['untouched'] );
+	}
+
+	/**
+	 * Ensure embed_links handles WP_Error objects returned by dispatch
+	 *
+	 * @ticket 56566
+	 */
+	public function test_link_embedding_returning_wp_error() {
+		$return_wp_error = function() {
+			return new WP_Error( 'some-error', 'This is not valid!' );
+		};
+		add_filter( 'rest_pre_dispatch', $return_wp_error );
+
+		$response = new WP_REST_Response();
+		$response->add_link( 'author', rest_url( 'test' ), array( 'embeddable' => true ) );
+
+		$data = rest_get_server()->response_to_data( $response, true );
+
+		$this->assertArrayHasKey( '_links', $data );
+		$this->assertSame( 1, did_filter( 'rest_post_dispatch' ) );
+		$this->assertSame( 'some-error', $data['_embedded']['author'][0]['code'] );
+
+		remove_filter( 'rest_pre_dispatch', $return_wp_error );
```

Comment:
We can remove this line. The teardown methods of each test case take care of resetting hooks.

Reply:
Ah, good to know! Thanks, @TimothyBJacobs, I've removed that in 08f94ba.